pkg/utils/o11y/controller.go

```diff
@@ -83,3 +90,97 @@ func (h *SuiteController) GetRegexPodNameWithSize(podNameRegex string, results [
 
 	return podNameWithSize, nil
 }
+
+func (h *SuiteController) QuayImagePushPipelineRun(quayOrg, secret, namespace string) (*v1beta1.PipelineRun, error) {
+
+	pipelineRun := &v1beta1.PipelineRun{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "tekton.dev/v1beta1",
+			Kind:       "PipelineRun",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "pipeline-egress-",
+			Namespace:    namespace,
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			PipelineSpec: &v1beta1.PipelineSpec{
+				Tasks: []v1beta1.PipelineTask{
+					{
+						Name: "buildah-quay",
+						TaskRef: &v1beta1.TaskRef{
```

Comment:
Any reason why you chose not to embed a `taskSpec` instead ([example](https://github.com/tektoncd/pipeline/blob/release-v0.46.x/examples/v1beta1/pipelineruns/pipelinerun-with-pipelinespec-and-taskspec.yaml#L9))? That way you shouldn't need to create or delete a `Task`.

Reply:
There is no existing function to create a pipeline run that includes taskSpec.

Reply:
Updated